delta/sourcing/src/main/scala/ch/epfl/bluebrain/nexus/sourcing/SnapshotStrategyConfig.scala

```diff
@@ -0,0 +1,52 @@
+package ch.epfl.bluebrain.nexus.sourcing
+
+import ch.epfl.bluebrain.nexus.sourcing.SnapshotStrategy.SnapshotPredicate
+
+/**
+  * A Snapshot configuration that will be made every numberOfEvents and keepNSnapshots will be kept
+  * deleteEventsOnSnapshot allows to delete events prior which are older than
+  * the older snapshot
+  *
+  * @param numberOfEvents         the optional frequency we need to trigger snapshots
+  * @param keepNSnapshots         the optional number of snapshots we need to keep
+  * @param deleteEventsOnSnapshot the optional flag to decide if we need to delete old events
+  */
+final case class SnapshotStrategyConfig private (
```

Comment:
I would split the config into an ADT.

And are there cases where we don't want to snapshot every now and then ? It would be even simpler

Reply:
Configurations should be simple and they should not mirror the actual thing. The ADT just adds complexity in the pureconfig reader side.

Reply:
You get at least part of the complexity here.

And I think we should fail if one of the property is not defined, not falling back to NoSnapshot

Reply:
@imsdu this is already the case, see the companion object (notice that class is private, so it will be constructed through the companion object)